engine/src/main/java/org/terasology/engine/particles/updating/ParticleUpdaterImpl.java

```diff
@@ -125,7 +125,8 @@ public void initialize() {
 
         for (Class<?> type : environment.getTypesAnnotatedWith(RegisterParticleSystemFunction.class)) {
             if (!ParticleSystemFunction.class.isAssignableFrom(type)) {
-                logger.error("Cannot register particle system function {}, must be a subclass of ParticleSystemFunction", type.getSimpleName());
+                logger.atError().addArgument(() -> type.getSimpleName()).
+                        log("Cannot register particle system function {}, must be a subclass of ParticleSystemFunction");
```

Comment:
`getSimpleName()` should be O(1) so impact neglectable
```suggestion
                logger.error("Cannot register particle system function {}, must be a subclass of ParticleSystemFunction", type.getSimpleName()); //NOPMD
```